tests/integration/migrator/conftest.py

```diff
@@ -21,15 +23,21 @@ def project_root_path_fixture(shared_datadir: Path) -> Path:
 
 
 @pytest.fixture(name="migrator_builder")
-def migrator_builder_fixture(devnet_gateway_url: str, project_root_path: Path):
-    migrator_builder = Migrator.Builder(
-        migrator_execution_environment_builder=MigratorExecutionEnvironment.Builder(),
-        gateway_facade_builder=GatewayFacade.Builder(project_root_path),
+def migrator_builder_fixture():
+    exc_env_builder = MigratorExecutionEnvironment.Builder()
+    builder = Migrator.Builder(migrator_execution_environment_builder=exc_env_builder)
+    builder.set_migration_execution_environment_config(
+        MigratorExecutionEnvironment.Config(
+            signer=PatchedStarkCurveSigner(1, KeyPair(1, 2), 2), token=None
+        )
     )
+    return builder
 
-    migrator_builder.set_network(devnet_gateway_url)
 
-    return migrator_builder
+@pytest.fixture(name="gateway_facade")
+def gateway_facade_fixture(devnet_gateway_url, project_root_path):
+    client = GatewayClient(devnet_gateway_url, chain=StarknetChainId.TESTNET)
+    return GatewayFacade(project_root_path=project_root_path, gateway_client=client)
```

Comment:
what do you think about making `Migrator.Builder` accept `gateway` via `set_gateway` method and therefore let tests forget about having to import `gateway_facade` fixture?

from application code perspective this is indifferent as far as I see

Reply:
seems reasonable

Reply:
i will make the builder store the intermediate value so calls can be executed in any order.